lib/guard.js

```diff
@@ -13,32 +13,53 @@ module.exports = function makeGuard(config) {
   // flatten agents
   config.groups
     .forEach(function (group) {
-      var rules = group.rules
-        .filter(function (rule) {
-          return !!rule.path;
-        })
-        .map(function (rule) {
-          return {
-            pattern: patterns.path(rule.path),
-            allow: rule.rule.toLowerCase() !== 'disallow'
-          };
-        })
+      const accessibilityRules = group.rules
+        .filter(({ rule, path }) => !!path && ['allow', 'disallow'].includes(rule.toLowerCase()))
+        .reduce((acc, { rule, path }) => {
+          const repeatedPathIndex = acc.findIndex((element) => {
```

Comment:
Don't use the index. The elements are objects, so you can just get a reference to it like this:
```js
const existingRuleSpec = acc.findIndex(rule => rule.path === path);
```
and mutate it if it exists.